lib/features/forest/config.dart

```diff
@@ -65,17 +101,115 @@ class ForestConfigState extends ConsumerState<ForestConfig> {
               pageControllerProvider:
                   forestPageControllerProvider, // Optional navigation
 
-              onPressed: () {
-                rSource(
-                  context,
-                  ref,
-                  ['model_template', 'model_build_random_forest'],
-                );
+              onPressed: () async {
+                selectedAlgorithm == AlgorithmType.traditional
+                    ? await rSource(
+                        context,
+                        ref,
+                        ['model_template', 'model_build_random_forest'],
+                      )
+                    : await rSource(
+                        context,
+                        ref,
+                        ['model_template', 'model_build_conditional_forest'],
+                      );
+
+                await ref.read(forestPageControllerProvider).animateToPage(
+                      // Index of the second page.
+                      1,
+                      duration: const Duration(milliseconds: 300),
+                      curve: Curves.easeInOut,
+                    );
               },
               child: const Text('Build Random Forest'),
             ),
             configWidgetGap,
             Text('Target: ${getTarget(ref)}'),
+            configWidgetGap,
```

Comment:
Please follow the same style of layout where possible - so a blank line around the gap - this is consistent everywhere please.